compiler/include/misc.h

```diff
@@ -70,7 +70,10 @@
 #define USR_STOP       exitIfFatalErrorsEncountered
 
 // INT_ASSERT is intended to become no-op in production builds of compiler
-#define INT_ASSERT(x) do { if (!(x)) INT_FATAL("assertion error"); } while (0)
+#define SELECT_ASSERT(_1, _2, NAME) NAME
+#define INT_ASSERT(...) SELECT_ASSERT(__VA_ARGS__, INT_ASSERT2, INT_ASSERT1)(__VA_ARGS__)
+#define INT_ASSERT1(x) do { if (!(x)) INT_FATAL("assertion error"); } while (0)
+#define INT_ASSERT2(x, s) do { if (!(x)) INT_FATAL((s), "assertion error"); } while (0)
```

Comment:
Thanks for tackling this, @chris980!

I think that most of the USR/INT macros take the source location as their first optional argument and then the stuff that's specific to the macro after that, so would suggest swapping the order of s and x.

I'm also trying to think how to test this...  I wonder if we could find any tests in the test system whose .bad files have internal assertion failures, and if so, whether they would be candidates for attaching a location like this...